Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -647,23 +636,51 @@ When the sources are not available, one can decompile the app and check the jar
+- V7.3: "Debugging symbols have been removed from native binaries."
+- V7.4: "Debugging code has been removed, and the app does not log verbose errors or debugging messages."
 - V7.5: "All third party components used by the mobile app, such as libraries and frameworks, are identified, and checked for known vulnerabilities."
+- V7.6: "The app catches and handles possible exceptions."
+- V7.7: "Error handling logic in security controls denies access by default."
+- V7.8: "In unmanaged code, memory is allocated, freed and used securely."
+- V7.9: "Free security features offered by the toolchain, such as byte-code minification, stack protection, PIE support and automatic reference counting, are activated."
+
+#### CWE
 
-##### CWE
+- CWE-20 - Improper Input Validation
+- CWE-215 - Information Exposure through Debug Information
+- CWE-388 - Error Handling
+- CWE-489 - Leftover Debug Code
+- CWE-656 - Reliance on Security through Obscurity
 - CWE-937 - OWASP Top Ten 2013 Category A9 - Using Components with Known Vulnerabilities
 
-##### Tools
-- [Black Duck](https://www.blackducksoftware.com/ "Black Duck")
-- [Sourceclear](https://www.sourceclear.com/ "Sourceclear")
-- [Snyk](https://snyk.io/ "snyk")
-- [Gradle license plugn](https://github.com/hierynomus/license-gradle-plugin "Gradle license plugin")
-- [Dependency-check-gradle](https://github.com/jeremylong/dependency-check-gradle "Dependency-check-gradle")
-- [MobSF](https://www.github.com/MobSF/Mobile-Security-Framework-MobSF, "MobSF")
+#### Tools
+
+- ProGuard - https://www.guardsquare.com/en/proguard
+- jarsigner - http://docs.oracle.com/javase/7/docs/technotes/tools/windows/jarsigner.html
+- Xposed - http://repo.xposed.info/
+- Drozer - https://labs.mwrinfosecurity.com/assets/BlogFiles/mwri-drozer-user-guide-2015-03-23.pdf
+- GNU nm - https://ftp.gnu.org/old-gnu/Manuals/binutils-2.12/html_node/binutils_4.html
+- Black Duck - https://www.blackducksoftware.com/
+- Sourceclear -  https://www.sourceclear.com/
+- Snyk - https://snyk.io/
+- Gradle license plugn - https://github.com/hierynomus/license-gradle-plugin
+- Dependency-check-gradle - https://github.com/jeremylong/dependency-check-gradle
+- MobSF - https://www.github.com/MobSF/Mobile-Security-Framework-MobSF
+- Squares leak canary - https://github.com/square/leakcanary
+- Memory Profiler from Android Studio - https://developer.android.com/studio/profile/memory-profiler
+- Android Java Deserialization Vulnerability Tester - https://github.com/modzero/modjoda
+
+#### Memory Analysis References
+- [A brief history of Android deserialization vulnerabilities](https://lgtm.com/blog/android_deserialization "android deserialization")
+- [9 ways to avoid memory leaks in Android](https://android.jlelse.eu/9-ways-to-avoid-memory-leaks-in-android-b6d81648e35e "a blog")
+- [Memory Leak Patterns in Android](https://android.jlelse.eu/memory-leak-patterns-in-android-4741a7fcb570 "a blog")

```

Comment:
same as above. Please use a proper title instaed of a blog

Reply:
fixed